src/utils/index.ts

```diff
@@ -0,0 +1,156 @@
+import type { InterpolateOptions, Path, PlainObject, Predicate, VeggiesError } from '../types.js'
+
+export const isNumber = (n: unknown): n is number => Number.isFinite(n)
+
+export const isNullsy = (val: unknown): val is undefined | null => val === undefined || val === null
+
+export const isNotNullsy = <T>(val: unknown): val is T => val !== undefined && val !== null
```

Comment:
There's already a widely accepted term for "undefined | null": nullish.

As in [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing), [asignement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment), or even [MDN's definition of what's a nullish value](https://developer.mozilla.org/en-US/docs/Glossary/Nullish)